src/HalfIntegers.jl

```diff
@@ -561,6 +562,38 @@ ishalfinteger(::HalfIntegerOrInteger) = true
 ishalfinteger(::AbstractIrrational) = false
 ishalfinteger(::Missing) = missing
 
+"""
+    ishalfoddinteger(x)
+
+Test whether `x` is numerically equal to some half-odd-integer.
+
+# Examples
+
+```jldoctest
+julia> ishalfoddinteger(3.5)
+true
+
+julia> ishalfoddinteger(2)
+false
+
+julia> ishalfoddinteger(1//3)
+false
+```
+"""
+ishalfoddinteger(x) = _isodd(twice(x))
+ishalfoddinteger(x::Rational) = denominator(x) == 2
+ishalfoddinteger(::Integer) = false
+ishalfoddinteger(::AbstractIrrational) = false
```

Comment:
I think this is not necessarily correct, because nothing prevents a user from defining an `AbstractIrrational` subtype that is equal to a rational number. The `AbstractIrrational` docstring even mentions this possibility:

>   If a subtype is used to represent values that may occasionally be rational (e.g. a square-root type that represents `√n` for integers `n` will give a rational result when `n` is a perfect square), then it should also implement `isinteger`, `iszero`, `isone`, and `==` with Real values (since all of these default to `false` for `AbstractIrrational` types), as well as defining `hash` to equal that of the corresponding Rational.

Since we cannot add `ishalfoddinteger` to that list, maybe we should let it fall back to the generic `_isodd(twice(x))`? Or should we just add a note to the `ishalfoddinteger` docstring?

Reply:
Ah, I see. I added the method referring to the `Base.isinteger` method.

https://github.com/JuliaLang/julia/blob/9c6fec13ed3a106a4c1e30d07d55eaa3701b5a59/base/irrationals.jl#L143

I now think we can just drop the method definition, and `ishalfinteger(x::Number)` can handle it.